library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608TrackRenderer.java

```diff
@@ -334,21 +477,32 @@ private String getDisplayCaption() {
       return null;
     }
 
+    // Every styling span should be closed at end of line
+    closeOpenSpans( STYLE_PRIORITY.STYLE_PRIORITY_ALL );
```

Comment:
There are issues present, among others:
- Paint On and Roll Up captions does not make too much sense without correct positioning as continuous movement renders text hardly readable. (Positioning is an upcoming change-set after these).
- Erase commands are not implemented correctly, so captions might stay on the screen longer than desired.
- Carriage returns are handled with adding line breaks to the text, leading to incorrect positioning. Line breaks do not match the behavior of the standard, as it defines rows of (somewhat independent) characters, line breaks are artificial handling of rows.
All these will be handled in an upcoming change.

I can update these lines to handle Paint On and Roll Up captions more correctly without positioning, unfortunately, I have never tested these as Paint On samples are unreasonable, while Roll Up samples tend to not have formatting.